tests/tests/bitasset_tests.cpp

```diff
@@ -1103,22 +1111,30 @@ BOOST_AUTO_TEST_CASE( hf_935_test )
          blocks += generate_blocks(db.get_dynamic_global_properties().next_maintenance_time, true, skip);
       }
 
-      // after hard fork 935, the limit order should be filled
+      // after hard fork 935, the limit order is filled only for the MSSR test
+      if( db.get_dynamic_global_properties().next_maintenance_time > HARDFORK_CORE_935_TIME &&
+          db.get_dynamic_global_properties().next_maintenance_time <= HARDFORK_CORE_1270_TIME )
       {
          // check median
          BOOST_CHECK( usd_id(db).bitasset_data(db).current_feed.settlement_price == current_feed.settlement_price );
-         if( i % 2 == 0 ) // MCR test, median MCR should be 350%
-            BOOST_CHECK_EQUAL( usd_id(db).bitasset_data(db).current_feed.maintenance_collateral_ratio, 3500 );
-         else // MSSR test, MSSR should be 125%
-            BOOST_CHECK_EQUAL( usd_id(db).bitasset_data(db).current_feed.maximum_short_squeeze_ratio, 1250 );
-         // the limit order should have been filled
-         // TODO FIXME this test case is failing for MCR test,
-         //            because call_order's call_price didn't get updated after MCR changed
-         // BOOST_CHECK( !db.find<limit_order_object>( sell_id ) );
-         if( i % 2 == 1 ) // MSSR test
-            BOOST_CHECK( !db.find<limit_order_object>( sell_id ) );
+         if( i % 2 != 0 ) { //  MSSR test, MSSR should be 125%, order filled
+            BOOST_CHECK_EQUAL(usd_id(db).bitasset_data(db).current_feed.maximum_short_squeeze_ratio, 1250);
+            BOOST_CHECK(!db.find<limit_order_object>(sell_id));
+         }
+         // go beyond hard fork 1270
+         blocks += generate_blocks(HARDFORK_CORE_1270_TIME, true, skip);
```

Comment:
To be more accurate, here should be
```
         blocks += generate_blocks(HARDFORK_CORE_1270_TIME - mi, true, skip);
         blocks += generate_blocks(db.get_dynamic_global_properties().next_maintenance_time, true, skip);
```

Reply:
thank you for the suggestion, i actually tried to make it like that everywhere but there are some problems not sure why.

in the case of this particular test(`hf_935_test`) when i make the generate block chains i get error:

```
83898ms th_a       bitasset_tests.cpp:1150       test_method          ] 10 assert_exception: Assert Exception
prev: popping block would leave head block null
    {}
    th_a  fork_database.cpp:41 pop_block

    {}
    th_a  db_block.cpp:477 pop_block
```